Fix for error on Chromium UIA combo-box when restoring Advanced Settings

CyrilleB79 opened the pull request:
### Link to issue number:
Fixes #12299

### Summary of the issue:
Using .Selection on a combo-box rather than .GetSelection was causing an error when restoring Advanced settings to default.

### Description of how this pull request fixes the issue:

Modified .Selection to .GetSelection()

### Testing strategy:
Tested restoration with and without the advanced settings panel enabled.
The window is correctly closed when pressing "OK".

### Known issues with pull request:
During these tests, I have identified another path that may cause the settings not to be saved. Issue coming soon.
This PR only fixes the combo-box .Selection error that was probably introduced when with Chromium UIA feature.
It does not fixes the issue of advanced settings not saved correctly (probably introduced by Wx 4.1.1)

### Change log entry:
None: alpha regression

### Code Review Checklist:

This checklist is a reminder of things commonly forgotten in a new PR.
Authors, please do a self-review and confirm you have considered the following items.
Mark items you have considered by checking them.
You can do this when editing the Pull request description with an x: `[ ]` becomes `[x]`.
You can also check the checkboxes after the PR is created.

- [x] Pull Request description is up to date.
- [x] Unit tests.
- [x] System (end to end) tests.
- [x] Manual tests.
- [x] User Documentation.
- [x] Change log entry.
- [x] Context sensitive help for GUI changes.

CyrilleB79 committed:
Fix for error on Chromium UIA combo-box when restoring Advanced Settings.

CyrilleB79 commented:
Cc @seanbudd, @XLTechie 
Sorry, seems that you already were assigned to / investigating / working on #12299; I have missed some of your comments.
Note that I do not reproduce the secondary issue indicated by @XLTechie in #12299.
@XLTechie is this issue still present with the build of this PR on your side?

XLTechie commented:
Neither issue occurs in the PR build. Well done.

I will however note, as I think you know, that pressing Restore Defaults while 
the checkbox is unchecked, does not restore the defaults.
That is definitely not the expected behavior, but I don't know how long it has 
been the case. It probably preceded the bug this PR solves.

CyrilleB79 commented:
Regarding the other issue (let's call it the "Visually modified options not saved"), I describe it here to keep a track. I cannot open a separate issue for now since there is no version of NVDA where it occurs since the error of #12299 happens before and since Wx 4.1.1 was introduced after Chromium UIA option.

STR:
1. Work on a fresh version of the snapshot of this PR (to have the GetSelection issue fixed).
2. Open advanced settings panel
3. Check the checkbox to enable the panel
4. Modify each option (e.g. check the unchecked checkboxes, change selected items of combo-boxes, etc.)
5. Press OK
6. Open again the advanced settings panel
7. Press the "Restore defaults" button (witout having activated the panel first)
8. Press OK
9. Open Advanced settings panel
10. Check the checkbox to enable the panel
11. Tab through the various options

Expected behaviour:
At step 11, the options are the default options.

Actual behaviour:
At step 11, the options remained the modified options.
Note that at step 7, the values of combo-boxes and the numeric spin control are visually modified (and probably also the checkboxes but I am not able to guarantee it visually due to low vision). However these values seem not to be taken into account at step 8 when validating the config.

Note:
I have taken last Python 3.7 snapshot (nvda_snapshot_alpha-21877,8e4b189b) and added the fix of this PR. In this modified build, the "Visually modified options not saved" issue is not present.

@seanbudd, there are 2 options:
1. Merge quickly the fix of this simple PR to close #12299 and open a new issue for the "Visually modified options not saved"
2. Keep this PR open until the "Visually modified options not saved issue" is also fixed.
Let me know what you prefer.

Personally, I prefer the first option, because:
* The error in #12299 and the "Visually modified option not saved" issue have two different causes and have not been introduced at the same moment.
* Should we downgrade wxPython (I hope not!), the "Visually modified selection not saved" bug would go away whereas the error of #12299 would remain.

OzancanKaratas reviewed (approved):
This is great! It worked as expected. @seanbudd, please review.

seanbudd commented on source/gui/settingsDialogs.py:
```diff
@@ -2815,7 +2815,7 @@ def haveConfigDefaultsBeenRestored(self):
 			and self.winConsoleSpeakPasswordsCheckBox.IsChecked() == self.winConsoleSpeakPasswordsCheckBox.defaultValue
 			and self.cancelExpiredFocusSpeechCombo.GetSelection() == self.cancelExpiredFocusSpeechCombo.defaultValue
-			and self.UIAInChromiumCombo.selection == self.UIAInChromiumCombo.defaultValue
+			and self.UIAInChromiumCombo.GetSelection == self.UIAInChromiumCombo.defaultValue
```
```suggestion
			and self.UIAInChromiumCombo.GetSelection() == self.UIAInChromiumCombo.defaultValue

```

It should be this, right?

seanbudd reviewed (commented):
@CyrilleB79 - no worries at all for taking this up, I was planning on starting this investigation today and I am glad that you've done the work. 

After a quick test, the change I've suggested seems to fix the "Visually modified options not saved issue" but I will do some further testing to confirm.

CyrilleB79 committed:
Fix missing ()

Co-authored-by: Sean Budd <seanbudd123@gmail.com>

CyrilleB79 commented:
Oh my bad!
No "Visual modification not saved" bug; just missing parenthesis that were causing `self.advancedControls.haveConfigDefaultsBeenRestored()` test to fail each time...
Thanks @seanbudd, I have committed your suggestion. This PR should be OK now, saving the advanced settings panel works as expected.

CyrilleB79 committed:
Linting

seanbudd reviewed (approved):
Thanks @CyrilleB79
